### ISSUE TITLE
onMapReady never called when getMapAsync is posted

### ISSUE THREAD

**Issue by kjkrum** [mapbox/mapbox-gl-native#9923]
Using Mapbox Android SDK 5.1.2.  I've extended `SupportMapFragment`.  Originally I was calling `getMapAsync` in `onCreate`:

```Java
@Override
public void onCreate(@Nullable final Bundle savedInstanceState) {
	super.onCreate(savedInstanceState);
	setHasOptionsMenu(true); // this is relevant
	if(savedInstanceState == null) {
		getMapAsync(new OnMapReadyCallback() {
			@Override
			public void onMapReady(final MapboxMap map) {
				// set up map
			}
		});
	}
}
```

The problem was that `onMapReady` was sometimes being called before the fragment's `onCreateOptionsMenu`, resulting in an NPE when I tried to access a menu item while setting up the map.  So I tried posting the call to `getMapAsync`.  When I do this, `onMapReady` is never called.

```Java
private final Handler mHandler = new Handler(Looper.getMainLooper());

@Override
public void onCreate(@Nullable final Bundle savedInstanceState) {
	super.onCreate(savedInstanceState);
	setHasOptionsMenu(true); // this is relevant
	if(savedInstanceState == null) {
		mHandler.post(new Runnable() {
			public void run() {
				getMapAsync(new OnMapReadyCallback() {
					@Override
					public void onMapReady(final MapboxMap map) {
						// set up map
					}
				});
			}
		});
	}
}
```
So I tried posting from inside `onMapReady`.  When I do this, everything works as expected.

```Java
private final Handler mHandler = new Handler(Looper.getMainLooper());

@Override
public void onCreate(@Nullable final Bundle savedInstanceState) {
	super.onCreate(savedInstanceState);
	setHasOptionsMenu(true); // this is relevant
	if(savedInstanceState == null) {
		getMapAsync(new OnMapReadyCallback() {
			@Override
			public void onMapReady(final MapboxMap map) {
				mHandler.post(new Runnable() {
					public void run() {
						// set up map
					}
				});
			}
		});
	}
}
```
When checking for duplicate issues, I ran across #9136.  I wonder if the fix for that created this.

**Comment by tobrun** [comment]
Thank you for the report, will look into reproducing this. fwiw if you want to add additional logic I would rather advice not extending SupportMapFragment but extending SupportFragment instead and copying over the SupportMapFragment implementation. 

re. #9136, the PR fixing that issue hasn't been released yet. 
I have added the related milestone to the ticket to reflect it.

**Comment by tobrun** [comment]
This has been fixed with #10717
